sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -185,4 +214,77 @@ private ChatEvent parsePushNotificationEvent(ChatEventType chatEventType, ChatPu
         this.logger.info(" Result: " + jsonString);
         return NotificationUtils.toEventPayload(chatEventType, jsonString);
     }
+
+    private void scheduleRegistrationRenew(final long delayMs, final int tryCount) {
+        if (!this.isPushNotificationsStarted) {
+            this.logger.info("Push notification has already been stopped! No need to renew registration.");
+            return;
+        }
+
+        if (this.deviceRegistrationToken == null) {
+            stopPushNotifications();
+            Throwable throwable = new Throwable("No device registration token stored!");
+            this.logger.logThrowableAsError(throwable);
+            if (this.registrationErrorHandler != null) {
+                this.registrationErrorHandler.accept(throwable);
+            }
+            return;
+        }
+
+        if (tryCount > NotificationUtils.MAX_REGISTRATION_RETRY_COUNT) {
+            stopPushNotifications();
+            Throwable throwable = new Throwable(
```

Comment:
I wonder if using `Throwable` for these cases is a good practice //cc @srnagar, @JonathanGiles.

Reply:
I construct a Throwable here to call this.registrationErrorHandler.accept(throwable);
It's a bit strange indeed. Is there better approach?

Reply:
Create an exception instead. The error handler handles all throwables (including Exception and Error).